src/Context/filter-context.jsx

```diff
@@ -0,0 +1,72 @@
+import { createContext, useContext, useReducer } from 'react';
+
+const filterContext = createContext();
+
+const initFilterState = {
+	sortBy: '',
+	price: 0,
+	gender: '',
+	categories: [],
+	rating: 1,
+};
+const sortByPrice = (productList, sortBy) => {
+	if (sortBy === 'LOW_TO_HIGH')
+		return productList.sort((a, b) => a.discountPrice - b.discountPrice);
+	else if (sortBy === 'HIGH_TO_LOW')
+		return productList.sort((a, b) => b.discountPrice - a.discountPrice);
+	else return productList;
+};
+
+const filterByCategory = (productList, filterCategories) => {
```

Comment:
you can move these functions in a separate file like helpers

Reply:
I'll refactor the code in another PR